pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -33,10 +33,14 @@ typedef DependencyMetadata = Map<String, Metadata>;
 /// https://github.com/dart-lang/native/issues/1319
 class NativeAssetsBuildRunner {
   final Logger logger;
+  final void Function(String) progressLogger;
```

Comment:
Why do we need a different logger? To have the semantics that every previous message overwrites the previous one?

Also, why don't we make it a stream instead of a callback?

Reply:
We need a different logger, as the logger is intended for the logging of the build scripts, not of the progress of the building itself.

I like callbacks, they are simpler... :)

Reply:
Hm, and then we use two abstractions, maybe we should have a `final Logger progressLogger` instead? Basically it has the same callback function (but also a level) per message...